Add Jenkins 2.387.1 and 2.375.4

MarkEWaite opened the pull request:
## Add Jenkins 2.387.1 and 2.375.4

Jenkins LTS release included both releases on the same day.  Security advisory was published for releases prior to those two releases.

https://www.jenkins.io/security/advisory/2023-03-08/ is the advisory

There will be no further releases on the 2.375.x line (no updates to Jenkins 2.375.4).  I think that means it should be marked as "end of life", though I'm not sure.  If a reviewer comments on the expectations there, I'm happy to mark the 2.375 line as end of life.

MarkEWaite committed:
[jenkins] Declare the 2.387 LTS and mark 2.375 as EOL (#2695)

Jenkins LTS release included both releases on the same day. Security
advisory was published for releases prior to those two releases.

https://www.jenkins.io/security/advisory/2023-03-08/ is the advisory.

marcwrobel commented on products/jenkins.md:
```diff
@@ -27,12 +27,20 @@ releases:
     eol: false
     latest: "2.375.4"
-    latestReleaseDate: 2023-03-06
+    latestReleaseDate: 2023-03-08
```
This change will be reverted by our automation.

Our automation is using the tag date, not the GitHub release date. It's wrong but, for the time being, this is not possible to manually "force" the date (unless we disable the automation).

Will have to check (not in this PR) if the GitHub releases automation script I wrote for adoptium/temurin could be used here. It could fix the issue.

MarkEWaite replied:
No problem.  The tag for security releases of Jenkins core are created on a private repository a few days before the actual release.  That allows the release to be staged privately without risking premature disclosure of the vulnerabilities that are being fixed.

marcwrobel commented on products/jenkins.md:
```diff
@@ -15,9 +15,10 @@ auto:
 # releaseCycle 2 catches weekly releases, other are LTS.
 # For an LTS :
-# - Its support date is its release date, meaning there will never be backport of features.
+# - Its end of support date is the 'x.y.1' release date. Most of the new features backported in LTS
```
@MarkEWaite, is this new comment better than the old one ?

MarkEWaite replied:
```suggestion
# - Its support begins with the 'x.y.1' release date. The `supportDate` value should use that date.  Most of the new features backported in LTS
```

Isn't that date the beginning of support, not the end?

marcwrobel replied:
No, it's the end of support date (the naming is confusing). I thought it was OK to "end" the  considering after the 'x.y.1' release most of the changes are fixes.

But looking at https://endoflife.date/jenkins I realize that the `support` fields must be removed, because the `activeSupportColumn` was set to `false` in #2511, so those dates are never displayed.

marcwrobel committed:
[jenkins] Clarify the rules for LTS

After the discussion on https://github.com/endoflife-date/endoflife.date/pull/2695#issuecomment-1472929098, it has been decided that LTS release dates must be the first patch version release date. For example release date of 2.375 is the release date of 2.375.1. Comments has been changed to reflect this new rule.

The support dates has also been removed because activeSupportColumn was disabled in #2511.

marcwrobel reviewed (approved):
LGTM now, merging (note: I observed today that the preview is not in sync with this PR's latest commit, even if it was re-deployed, strange...).

@MarkEWaite it would be great if you could review https://endoflife.date/jenkins and let me know if everything looks good to you now.
